samples/kvsWebRTCClientViewer.c

```diff
@@ -79,22 +129,37 @@ INT32 main(INT32 argc, CHAR *argv[])
     message.payloadLen = (buffLen / SIZEOF(CHAR)) - 1;
     message.correlationId[0] = '\0';
 
-    CHK_STATUS(signalingClientSendMessageSync(pSampleConfiguration->signalingClientHandle, &message));
+    retStatus = signalingClientSendMessageSync(pSampleConfiguration->signalingClientHandle, &message);
+    if(retStatus != STATUS_SUCCESS) {
+        printf("[KVS Viewer] signalingClientSendMessageSync(): operation returned status code: 0x%08x \n", retStatus);
+        goto CleanUp;
+    }
+
     // Block forever
     THREAD_SLEEP(MAX_UINT64);
 
 CleanUp:
     printf("[KVS Viewer] Cleaning up....\n");
-    CHK_LOG_ERR_NV(retStatus);
 
     if (locked) {
-        MUTEX_UNLOCK(pSampleConfiguration->sampleConfigurationObjLock);
+        pthread_mutex_unlock((pthread_mutex_t*) pSampleConfiguration->sampleConfigurationObjLock);
```

Comment:
There is an implementation of ptrheads for most platforms but this is similar to other PIC concepts - do we need to remove the portable public PIC api in favor of non-portable implementation?